pyEpiabm/pyEpiabm/sweep/update_place_sweep.py

```diff
@@ -20,25 +20,90 @@ def __call__(self, time: int):
         :type time: int
         """
         # Double loop over the whole population, clearing places
-        # and refilling them.
+        # of the variable population and refilling them.
+
+        # Can call a this line if being called in from file etc.
+        # place_params = Parameters.instance().place_params
         for cell in self._population.cells:
             for place in cell.places:
-                place.empty_place()
-                # Possibly want to later differentiate between
-                # the fixed population of a place ie staff, and
-                # the variable population ie customers.
-
-                # Ensure that the number of people put in the place
-                # is at most its capacity or the total number of
-                # people in the cell.
-                max_capacity = np.minimum(place.max_capacity,
-                                          len(place.cell.persons))
-
-                new_capacity = random.randint(1, max_capacity)
-                count = 0
-                while count < new_capacity:
-                    i = random.randint(1, len(place.cell.persons))
-                    # Checks person is not already in the place.
-                    if place.cell.persons[i-1] not in place.persons:
-                        place.add_person(place.cell.persons[i-1])
-                        count += 1
+                if place.place_type.value == 1:  # HOTEL
+                    place.empty_place()
+                    self.update_place_group(place)
+                    # not sure how to handle this yet so
+                    # just going to completely update each time
+                    # as in the previous code
+                elif place.place_type.value == 1:  # CAREHOME
+                    if not place.initialised:
```

Comment:
can we have a separate initialise function maybe? Seems a waste to be running this check on every timestep, and presumably we won't be adding new places mid-simulation? Not a big cost in the grand scheme of things though

Reply:
Yea, started writing a separate place_initialise sweep but wasn't sure. Would make sense though so will do that.